### DISCUSSION TITLE
The page UI component cannot adapt to different computer screens

### DISCUSSION THREAD

**Discussion by xiaozhiagi** [zauberzeug/nicegui#1377] [Q&A]
### Question

On one computer, the UI component layout can be displayed normally, but on other computers with different screen sizes, it cannot adapt to the UI component layout and becomes messy. Here is an example:

```python

with ui.card().style("margin-left:25%;min-height:1200px;width:50%;"):
      ui.html("<br>")
      ui.label('数据集路径').style("font-size:18px;font-weight:700;")
    with ui.dialog() as dialog, ui.card().style("width:80%;height:500px;background-color:#E5ECF6;"):
        ui.label('Hello world!')
        ui.button('Close', on_click=dialog.close)
    with ui.row().style("margin-left:5%;width:100%"):
        ui.input(label='数据缓存路径').style("width:78%")
        ui.button('选择路径', on_click=dialog.open).style("width:10%;margin-top:15px;")
````

**Comment by rodja** [comment]
Thanks for reaching out. Can you share some screenshots? What screen sizes are we talking about? It's hard to reproduce without this informations. Please take time to formulate your question. Quite a number of people will read and think about your message. Make it worth.

**Reply by xiaozhiagi** [reply]
Sorry, I can't upload pictures. May I know which public location is available for uploading pictures

**Reply by xiaozhiagi** [reply]
The main problem is that currently, width margin-left is only a percentage, but different computer screens cannot adapt to the page according to the percentage. For example, the button name of ui. button may collapse or wrap due to the small screen size

**Reply by rodja** [reply]
I'm sorry. I still do not understand your layout problem.

Btw: uploading of images can be done via drag/drop in the entry field here on GitHub. Or you can try our Discord server https://discord.gg/TEpFeAaF4f. There are more people helping out.
